media_kit/lib/src/libmpv/player.dart

```diff
@@ -74,13 +74,29 @@ class Player extends PlatformPlayer {
       calloc.free(aid);
       calloc.free(sid);
       calloc.free(no);
+
       // Raw [mpv_command] calls cause crash on Windows.
       final command = 'quit $code'.toNativeUtf8();
       _libmpv?.mpv_command_string(
         ctx,
         command.cast(),
       );
       calloc.free(command);
+
+      // Wake up in order to skip potentially hanging `mpv_wait_event()` within event loop.
+      _libmpv?.mpv_wakeup(ctx);
+
+      // Wait for shutdown event to actually occur. 
+      // If timed out, don't do anything. Perhaps implement a solution to handle this in the future.
+      await _mpvShutdownComplete.future.timeout(Duration(seconds: 3), onTimeout: () {});
+
+      // On android, calling `quit 0` does not actually clean the resources.
+      // We must call `mpv_destroy()` in order to prevent extremely high CPU usage.
+      // In the future, find out whether other platforms require this as well.
+      if(Platform.isAndroid){
+        _libmpv?.mpv_destroy(ctx);
```

Comment:
Doesn't seem to crash on macOS, if it's helpful for Android it should be helpful for other platforms too.